demo/webcam_demo_spatiotemporal_det.py

```diff
@@ -0,0 +1,803 @@
+                # displayed by the previous task
+                for frame_id in self.display_inds:
+                    frame = task.frames[frame_id]
+                    if self.show:
+                        cv2.imshow('Demo', frame)
+                        cv2.waitKey(int(1000 / self.output_fps))
+                    if self.video_writer:
+                        self.video_writer.write(frame)
+
+            cur_time = time.time()
+            logger.debug(
+                f'Display thread: {1000*(cur_time - start_time):.0f} ms, '
+                f'read id {read_id}, display id {display_id}')
+            start_time = cur_time
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        """Get data from read queue.
+
+        This function is part of the main thread.
+        """
+        if self.read_queue.qsize() == 0:
+            time.sleep(0.02)
+            return not self.stopped, None
+        else:
+            return self.read_queue.get()
+
+    def start(self):
+        """Start read thread and display thread."""
+        self.read_thread = threading.Thread(
+            target=self.read_fn, args=(), name='VidRead-Thread', daemon=True)
+        self.read_thread.start()
+        self.display_thread = threading.Thread(
+            target=self.display_fn,
+            args=(),
+            name='VidDisplay-Thread',
+            daemon=True)
+        self.display_thread.start()
+
+        return self
+
+    def clean(self):
+        """Close all threads and release all resources."""
+        self.stopped = True
+        self.read_lock.acquire()
+        self.cap.release()
+        self.read_lock.release()
+        self.output_lock.acquire()
+        cv2.destroyAllWindows()
+        if self.video_writer:
+            self.video_writer.release()
+        self.output_lock.release()
+
+    def join(self):
+        """Waiting for the finalization of read and display thread."""
+        self.read_thread.join()
+        self.display_thread.join()
+
+    def display(self, task):
+        """Add the visualized task to the display queue.
+
+        Args:
+            task (TaskInfo object): task object that contain the necessary
+            information for prediction visualization.
+        """
+        with self.display_lock:
+            self.display_queue[task.id] = (True, task)
+
+    def get_output_video_writer(self, path):
+        """Return a video writer object.
+
+        Args:
+            path (str): path to the output video file.
+        """
+        return cv2.VideoWriter(
+            filename=path,
+            fourcc=cv2.VideoWriter_fourcc(*'mp4v'),
+            fps=float(self.output_fps),
+            frameSize=(self.display_size[1], self.display_size[0]),
+            isColor=True)
+
+
+class Visualizer:
+    """Tools to visualize predictions.
+
+    Args:
+        plate (str): The plate used for visualization.
+            Default: plate_blue.
```

Comment:
I don't quite understand this. Why is the initialization assignment at the bottom different from that in the docstring?

Reply:
plate define several colors (denoted by hex values in a string), which will be used for visualization. An alternative to make it less ambiguous is to define plate_blue as a global var.

Reply:
I forgot to replace `plate_blue` with the hex values. Will fix this

Reply:
@kennymckormick Could you please share some reference materials for the color plate? How did you get the blue plate and the green plate?

Reply:
I generate the color plates with https://coolors.co/. Blue and Green are just two color plates I feel good about.

Reply:
I'll add some docs about this color plate later.